File: core/src/main/java/com/mcreater/amclcore/model/oauth/LoginDeviceCodeErrorType.java
package com.mcreater.amclcore.model.oauth;

public enum LoginDeviceCodeErrorType {
    AUTHORIZATION_PENDING,
    SLOW_DOWN,
    EXPIRED_TOKEN,
    INVALID_GRANT;

    public static LoginDeviceCodeErrorType parse(String s) {
        return valueOf(LoginDeviceCodeErrorType.class, s.toUpperCase());
    }
}

File: core/src/main/java/com/mcreater/amclcore/model/oauth/LoginDeviceCodeErrorType.kt
package com.mcreater.amclcore.model.oauth

import java.util.*

enum class LoginDeviceCodeErrorType {
    AUTHORIZATION_PENDING,
    SLOW_DOWN,
    EXPIRED_TOKEN,
    INVALID_GRANT;

    companion object {
        @JvmStatic
        fun parse(s: String): LoginDeviceCodeErrorType {
            return enumValueOf<LoginDeviceCodeErrorType>(s.uppercase(Locale.getDefault()))
        }
    }
}